src/api/export.ts

```diff
@@ -176,31 +200,47 @@ async function _export({
 				// parse link rel=preload headers and embed them in the HTML
 				let link = parseLinkHeader(r.headers.get('Link') || '');
 				link.refs.forEach((ref: Ref) => {
-					if (ref.rel === 'preload') {
+					if (ref.rel === 'preload' && typeof(body) === 'string') {
 						body = body.replace('</head>',
 							`<link rel="preload" as=${JSON.stringify(ref.as)} href=${JSON.stringify(ref.uri)}></head>`)
 					}
 				});
 
 				if (pathname !== '/service-worker-index.html') {
-					const cleaned = clean_html(body);
-
-					const base_match = /<base ([\s\S]+?)>/m.exec(cleaned);
-					const base_href = base_match && get_href(base_match[1]);
-					const base = resolve(url.href, base_href);
-
-					let match;
-					let pattern = /<a ([\s\S]+?)>/gm;
-
-					while (match = pattern.exec(cleaned)) {
-						const attrs = match[1];
-						const href = get_href(attrs);
+					if(typeof(body) === 'string') {
```

Comment:
you could combine this with the previous `if` check to remove a level of indentation, which would make things easier to read:

    if (pathname !== '/service-worker-index.html' && typeofbody === 'string') {